### PR TITLE
RDCC 1538 - Changes to GetUsers EndPoint as requested by FPLA

### PR DESCRIPTION
**Pull request by pksaxena01**
### JIRA link (if applicable) ###
RDCC-1538

### Change description ###
Get Users end point calls to sidam for every user profile to fetch roles. FPLA does not need this roles to be fetched hence to improve the performance, we have introduced a param to filter the IDAM call

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/ProfessionalUserInternalController.java [hunk 1]
```diff
@@ -1,196 +1,196 @@
...
+@RequestMapping(
+        path = "refdata/internal/v1/organisations",
+        produces = APPLICATION_JSON_VALUE
+)
+@Validated
+@RestController
+@Slf4j
+public class ProfessionalUserInternalController extends SuperController {
+
+    @ApiOperation(
```
Comment (KOTLAPRA23) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/ProfessionalUserInternalController.java:47]: why entire class is showing difference here.could you please revert the file and add only the changes that you modified

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/FindUsersByOrganisationTest.java [hunk 2]
```diff
@@ -1,283 +1,298 @@
...
+        bearerTokenForExternalRole = professionalApiClient.getMultipleAuthHeadersExternal(role, firstName, lastName, userEmail);
+
+        NewUserCreationRequest userCreationRequest = aNewUserCreationRequest()
+                .firstName(firstName)
+                .lastName(lastName)
+                .email(userEmail)
+                .roles(userRoles)
+                .jurisdictions(createJurisdictions())
+                .build();
+        professionalApiClient.addNewUserToAnOrganisation(orgIdentifierResponse, hmctsAdmin, userCreationRequest, HttpStatus.CREATED);
```
Comment (KOTLAPRA23) [inline src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/FindUsersByOrganisationTest.java:55]: same here also

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/ProfessionalUserInternalController.java (hunk 1 above)
Comment (pksaxena01) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/ProfessionalUserInternalController.java:47]: Please can you apply hidden whitespaces setting and review. I have tried by reverting the change but no effect. I will take care this next time. Thanks.

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/FindUsersByOrganisationTest.java (hunk 2 above)
Comment (pksaxena01) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/FindUsersByOrganisationTest.java:55]: Please can you apply hidden whitespaces setting and review. I have tried by reverting the change but no effect. I will take care this next time. Thanks.

**Pull request closed without merging**